src/utils/permissions.js

```diff
@@ -12,6 +12,13 @@ const hasPermission = (action) => {
       if (roleIndex !== -1) {
         permissions = rolePermissions[roleIndex].permissions;
       }
+      if(typeof(action) === 'object'){
+        for (const key of Object.keys(action)) {
+          if(userPermissions?.includes(key) || permissions?.includes(key)){
+            return true;
+          }
```

Comment:
This might have to be recursive to accommodate the other comment. I think there's a helper function in `PermissionsConst` which you could use for this.

Reply:
Didnt get you

Reply:
`PermissionsConst` involves nested subcategories (Projects > WBS > Tasks for example). It looks like this checks for any subpermission in a category, so it would have to also include recursion for nested categories if you were to merge `src/utils/constants` with `PermissionsConst`.

Reply:
Oh, perhaps the helper function I was thinking of though is in `PermissionListItem` instead.